plugin/vscode/performance-writer.js

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2020  Matthew Glazar
+// See end of file for extended copyright information.
+
+const fs = require('fs');
+const path = require("path");
+const process = require("process");
+
+class PerformanceWriter {
+    
+    constructor (extensionContext) {
+        this._dataAttributes = {
+            columns:'timestamp,file-ref,file-name-or-duration,event-id'
+        };
+        this._tmpLogLines = []
+        this._fileName = null;
+        this._fileIndex = null;
+        this._extensionContext = extensionContext;
+        fs.mkdirSync(this._extensionContext.logUri.fsPath)
+        this._performanceTxtPath = path.join(this._extensionContext.logUri.fsPath, "qljsPerformanceTrace.txt");
+        this._currentSessionMap = this._extensionContext.globalState.get('qljsFileMap');
+        this._writableStream = fs.createWriteStream(this._performanceTxtPath, {encoding: 'utf8', flags: 'a'})
+    }
+
+    createHeader (){
+        this.writeToTmpArray(null,"header");
+    }
+    
+    fileNameOptimizer (){
+        this._fileName = this._fileName.split(path.sep);
+        this._fileName = this._fileName[this._fileName.length - 1];
```

Comment:
Performance: Let's try to avoid string manipulation (especially allocation) in the hot path. I think we can store the full path instead of just the last component. Or, we can look up by `vscode.Document` instance (with a `Map`) (though this might get confusing with renames... unsure).

Reply:
See comment reply in [here](https://github.com/quick-lint/quick-lint-js/pull/244#discussion_r632243508).

Reply:
[Addressed in latest commit]
this._fileName is now an argument without separating the path to have only the filename